src/test/java/spoon/test/trycatch/TryCatchTest.java

```diff
@@ -442,4 +444,60 @@ public void testTryWithVariableAsResource() {
 		tryStmt.removeResource(ctResource);
 
 	}
+
+	@Nested
+	class AddCatcherAt {
+		@Test
+		void addsCatcherAtTheSpecifiedPosition() {
+			// contract: the catcher should be added at the specified position
+			// arrange
+			Factory factory = createFactory();
+
+			CtTry tryStatement = factory.createTry();
+
+			CtCatch first = factory.createCatch();
+			first.setParameter(factory
+					.createCatchVariable()
+					.setType(factory.Type().createReference(IOException.class)));
+
+			CtCatch second = factory.createCatch();
+			second.setParameter(factory
+					.createCatchVariable()
+					.setMultiTypes(List.of(
+							factory.Type().createReference(ExceptionInInitializerError.class),
+							factory.Type().createReference(NoSuchFieldException.class))));
+
+			CtCatch third = factory.createCatch();
+			third.setParameter(factory
+					.createCatchVariable()
+					.setType(factory.Type().createReference(NoSuchMethodException.class)));
```

Comment:
This is kind of a mouthful to read, and there is as far as I can tell unnecessary complexity in having a catcher that catches multiple types (I can't see how this influences insertion?).

I'd suggest something like this:

```suggestion
			CtCatch first = createCatch(factory, IOException.class);
			CtCatch second = createCatch(factory, ExceptionInInitializerError.class);
			CtCatch third = createCatch(factory, NoSuchMethodException.class);
```

With a helper like so:

```java
		private <T> CtCatch createCatch(Factory factory, Class<T> typeToCatch) {
			CtTypeReference<T> typeReference = factory.Type().createReference(typeToCatch);
			return factory.createCatch().setParameter(
				factory.createCatchVariable().setType(typeReference)
			);
		}
```

Reply:
> createCatch(Factory, Class<T> )

Any reason why we pass the same instance of `Factory` and not create a new one whenever we want to call `createCatch`?